conan/cache/db/references.py

```diff
@@ -62,106 +57,68 @@ def _set_clause(self, ref: ConanReference, path=None, timestamp=None, remote=Non
         placeholders = ', '.join(['?' for _ in range(len(self.columns))])
-        r = conn.execute(f'INSERT INTO {self.table_name} '
+        r = self._conn.execute(f'INSERT INTO {self.table_name} '
                          f'VALUES ({placeholders})',
                          [ref.reference, ref.rrev, ref.pkgid, ref.prev, path, remote, timestamp,
                           None])
         return r.lastrowid
 
-    def set_remote(self, conn, ref: ConanReference, remote):
+    def set_remote(self, ref: ConanReference, remote):
         where_clause = self._where_clause(ref)
-        mew_remote = f'"{remote}"' if remote else 'NULL'
+        new_remote = f'"{remote}"' if remote else 'NULL'
         query = f"UPDATE {self.table_name} " \
-                f"SET {self.columns.remote}={mew_remote} " \
+                f"SET {self.columns.remote}={new_remote} " \
                 f"WHERE {where_clause};"
-        r = conn.execute(query)
+        r = self._conn.execute(query)
         return r.lastrowid
 
-    def update(self, conn, old_ref: ConanReference, new_ref: ConanReference = None, new_path=None,
-               new_remote=None, new_build_id=None):
+    def update(self, old_ref: ConanReference, new_ref: ConanReference = None, new_path=None,
+               new_remote=None, new_timestamp=None, new_build_id=None):
         if not new_ref:
             new_ref = old_ref
-        timestamp = time.time()
         where_clause = self._where_clause(old_ref)
-        set_clause = self._set_clause(new_ref, path=new_path, timestamp=timestamp,
+        set_clause = self._set_clause(new_ref, path=new_path, timestamp=new_timestamp,
                                       remote=new_remote, build_id=new_build_id)
         query = f"UPDATE {self.table_name} " \
                 f"SET {set_clause} " \
                 f"WHERE {where_clause};"
-        r = conn.execute(query)
+        try:
+            r = self._conn.execute(query)
+        except sqlite3.IntegrityError:
+            raise ConanReferenceAlreadyExist(f"Reference '{new_ref.full_reference}' already exists")
         return r.lastrowid
 
-    def delete_by_path(self, conn, path):
+    def delete_by_path(self, path):
         query = f"DELETE FROM {self.table_name} " \
                 f"WHERE path = ?;"
-        r = conn.execute(query, (path,))
+        r = self._conn.execute(query, (path,))
         return r.lastrowid
 
-    def remove(self, conn, ref: ConanReference):
+    def remove(self, ref: ConanReference):
         where_clause = self._where_clause(ref)
         query = f"DELETE FROM {self.table_name} " \
                 f"WHERE {where_clause};"
-        r = conn.execute(query)
+        r = self._conn.execute(query)
         return r.lastrowid
 
-    def all(self, conn, only_latest_rrev=False):
+    def all(self, only_latest_rrev=False):
         if only_latest_rrev:
             query = f'SELECT DISTINCT {self.columns.reference}, ' \
```

Comment:
Why using `DISTINCT` if you are already doing `GROUP BY` the reference?

Reply:
Why selecting `MAX({self.columns.timestamp})` if you are ordering the timestamp DESC? Shouldn't it return only the latest already?

Reply:
What is the meaning of `{self.columns.prev} IS NULL` ? binary packages?

Reply:
Yes, totally, `GROUP BY` and `MAX({self.columns.timestamp})` are redundant. I'll remove them.
This method list only the different Conan references (name/version@user/channel) I'm going to change the name because it's confusing. That's why the `{self.columns.prev} IS NULL`, because it should not list all the binaries.